src/api/ppx_deriving.cppo.ml

```diff
@@ -640,43 +728,48 @@ let mapper =
         | None -> raise_errorf ~loc "Cannot locate deriver %s" name
       in
       begin match payload with
-      | PTyp typ -> deriver typ
+      | PTyp typ -> copy_deriver deriver typ
       | _ -> raise_errorf ~loc "Unrecognized [%%derive.*] syntax"
       end
     | { pexp_desc = Pexp_extension ({ txt = name; loc }, PTyp typ) } ->
       begin match lookup_internal_or_external name with
       | Some (Internal { core_type = Some deriver }) ->
-        Ast_helper.with_default_loc typ.ptyp_loc (fun () -> deriver typ)
+        Ast_helper.with_default_loc typ.ptyp_loc (fun () ->
+          copy_deriver deriver typ)
       | _ -> Ast_mapper.(default_mapper.expr) mapper expr
       end
     | _ -> Ast_mapper.(default_mapper.expr) mapper expr
   in
   let structure mapper items =
     match items with
     | { pstr_desc = Pstr_type(_, typ_decls) as pstr_desc ; pstr_loc } :: rest when
-        List.exists (fun ty -> has_attr "deriving" ty.ptype_attributes) typ_decls
+        List.exists (fun ty -> has_attr_current "deriving" ty.ptype_attributes) typ_decls
```

Comment:
I think it would make sense to shadow `has_attr` with `has_attr_current` at the beginning of `mapper`, so that these don't have to change.